internal/events/events_dynamic_test.go

```diff
@@ -0,0 +1,119 @@
+package events
+
+import (
+	"testing"
+
+	"github.com/Taraxa-project/taraxa-indexer/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDecodeEventDynamicDpos(t *testing.T) {
+	// Test RewardsClaimedEvent
+	rewardsClaimedTopic := "0x9310ccfcb8de723f578a9e4282ea9f521f05ae40dc08f3068dfad528a65ee3c7" // Example topic for RewardsClaimed event
+	rewardsClaimedData := "0x000000000000000000000000000000000000000000000000095e271c526c181b"  // Example data for RewardsClaimed event
+	rewardsClaimedLog := models.EventLog{
+		Address:          "0x00000000000000000000000000000000000000fe",
+		LogIndex:         0,
+		Removed:          false,
+		Topics:           []string{rewardsClaimedTopic, "0x000000000000000000000000fc43217e71ec0a1cc480f3d210cd07cbde7374ec", "0x000000000000000000000000fc43217e71ec0a1cc480f3d210cd07cbde7374ec"},
+		Data:             rewardsClaimedData,
+		TransactionHash:  "0xecd3243842f3fe4a0e8419a0ac85e4a6098bde7635de53bcad18e57f80cc6463",
+		TransactionIndex: 1,
+	}
+
+	{
+		name, decoded, err := DecodeEventDynamic(rewardsClaimedLog)
+		t.Log(name)
+		t.Log(decoded)
+		if err != nil {
```

Comment:
I think you must also check that decoded data have proper data, not only that it doesn't have an error

Reply:
I think the internal lib kinda takes care of that:

```
func (abi ABI) getArguments(name string, data []byte) (Arguments, error) {
	// since there can't be naming collisions with contracts and events,
	// we need to decide whether we're calling a method or an event
	var args Arguments
	if method, ok := abi.Methods[name]; ok {
		if len(data)%32 != 0 {
			return nil, fmt.Errorf("abi: improperly formatted output: %q - Bytes: %+v", data, data)
		}
		args = method.Outputs
	}
	if event, ok := abi.Events[name]; ok {
		args = event.Inputs
	}
	if args == nil {
		return nil, fmt.Errorf("abi: could not locate named method or event: %s", name)
	}
	return args, nil
}
```